### PR TITLE
Add support for Python 3.10 on CI

### PR DESCRIPTION
**Pull request by DonJayamanne**
Found 1 bug as a result of this PR

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -573,15 +488,6 @@ jobs:
           beakerx_kernel_java install
           conda install pytorch cpuonly -c pytorch
 
-      # Install PyTorch with pip
-      - name: Install PyTorch on Windows and Linux
-        if: matrix.python != 'conda' && matrix.python != 'noPython' && (matrix.os == 'windows-latest' || matrix.os == 'ubuntu-latest')
-        run: pip install torch==1.7.1+cpu torchvision==0.8.2+cpu torchaudio===0.7.2 -f https://download.pytorch.org/whl/torch_stable.html
```
Comment (DonJayamanne) [inline .github/workflows/build-test.yml:579]: None of these are used,
We have these for some old tests that relate to fancy tooltips for tensor varaibles (and that functionality is not in the extension at all).

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -317,20 +317,15 @@ jobs:
...
         # macOS runners are expensive, and we assume that Ubuntu is enough to cover the UNIX case.
         os: [ubuntu-latest]
         # More details on includes/excludes can be found here https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-additional-values-into-combinations
         # Basically with exclude, you can exclude any of the combinations from the result matrix.
         # & with include, you can include additional items to the result matrix.
-        exclude:
-          - os: ubuntu-latest # We don't care how users start Remote Jupyter as long as it works.
-            pythonVersion: '3.9'
-            python: conda
-            jupyter: remote
```
Comment (DonJayamanne) [inline .github/workflows/build-test.yml:331]: THis isn't required, as we're already including this seprately (was meant to be removed in the previous PR)

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -317,20 +317,15 @@ jobs:
       matrix:
         jupyter: [raw, local]
         python: [nonConda, conda]
-        pythonVersion: [3.9]
+        pythonVersion: ['3.9', '3.10']
```
Comment (rchiodo) [inline .github/workflows/build-test.yml:320]: I believe these are going to fail? Or maybe Ian hasn't submitted his new test yet.

**Review by rchiodo — APPROVED** [review]

File: .github/workflows/build-test.yml (hunk 3 above)
Comment (IanMatthewHuff) [reply .github/workflows/build-test.yml:320]: I would expect it to fail. At least it did for me on my local machine. Didn't try on CI yet though.

Comment (DonJayamanne) [reply .github/workflows/build-test.yml:320]: I have filed an issue and yes, it will fail for Python 3.10.
I'll skip the test if using Python 3.10

Comment (IanMatthewHuff) [reply .github/workflows/build-test.yml:320]: Test validated :). 
image
image

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**